### DISCUSSION TITLE
NULL record issue for UpdateOrCreate

### DISCUSSION THREAD

**Discussion by mahdi-reverted** [livewire/livewire#5367] [1. Help] [answered]
Hi when i fetch "User" table related with "Personal_information", first time "Personal_information" returns null data as the user is new. Then the user has to fill the "Personal_information" table data. This time UpdateOrCreate gets array for "Personal_information". But after saving, when i fetch "User" table related with "Personal_information" the "Personal_information" returns object this time. So UpdateOrCreate returns error as it needs array. I want to use single UpdateOrCreate to update or create. What I am using in Mount Method is:

>  $this->user = Auth::user();
>         $this->personal_information = $this->user->personal_information;
>         if (isset($this->personal_information)) {
>             $this->personal_information = $this->personal_information->toArray();
>         }

**Comment by joshhanley** [comment, marked as answer]
@suzondas I'm sorry I don't really follow what the issue is. Can you try and explain it a different way and show all relevant code (as I don't see your `updateOrCreate` code.

Thanks!

**Reply by mahdi-reverted** [reply]
Thanks
I sloved null record with these:
withDefault(function () {
            return new General_information();

**Reply by joshhanley** [reply]
Ok great! 🙂
